doc/Publishing.md

```diff
@@ -1,152 +1,215 @@
+   - [3.3 Update Future Milestones](#33-update-future-milestones)
+   - [3.4 Merge Website PRs](#34-merge-website-prs)
+   - [3.5 Publish GitHub Pages](#35-publish-github-pages)
+4. [Troubleshooting](#4-troubleshooting)
+   - [Failures During Publishing](#41-failures-during-publishing)
 
-## Pre-Release Steps
+## 1. Pre-Release Steps
 
-### Yarn Upgrade
+### 1.1 Yarn Upgrade
 
-In general, it is recommended to perform a `yarn upgrade` on the repository prior to a release to update the `yarn.lock`.
-The upgrade helps to:
+Perform a `yarn upgrade` on the repository prior to a release to update the `yarn.lock`. The upgrade helps to:
 
-- Better represents what adopters will pull during a release.
+- Better represent what adopters will pull during a release.
 - Validate dependencies with our declared version ranges.
 - Fix known security vulnerabilities from dependencies.
 
-In order to successfully perform a `yarn upgrade` one must:
+To perform the upgrade:
 
-- Perform a `yarn upgrade` at the root of the repository.
-- Fix any potential compilation errors, typing errors, and failing tests that may have been introduced.
+- Run `yarn upgrade` at the root of the repository.
+- Fix any compilation errors, typing errors, and failing tests.
+- Open a PR with the changes ([example](https://github.com/eclipse-theia/theia/pull/13423)).
 - Confirm licenses and wait for the "IP Check" to complete ([example](https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/9377)).
 
-### Announce Release
+### 1.2 Announce Release
 
-It is a good idea to give a heads-up to developers and the community some hours before a release.
-At the time of writing this is [GitHub Discussions](https://github.com/eclipse-theia/theia/discussions). Here is an [example](https://github.com/eclipse-theia/theia/discussions/13314).
+- Provide a heads-up to developers and the community a few hours before the release.
+- Use [GitHub Discussions](https://github.com/eclipse-theia/theia/discussions) for the announcement.
 
-### Localization
+  > Title: Eclipse Theia v{version}
+  >
+  > Hey everyone 👋,
+  >
+  > The Eclipse Theia v{version} release is happening today, starting around {time} CET.
+  > If you have any nearly-complete PRs that need to be included, let us know in this thread before we begin.
+  >
+  > We’ll post updates when the release starts and once it’s done.
+  > Please avoid merging any PRs during the release until we confirm it's over.
 
-The localization (`nls`) updates should be performed before a release ([example](https://github.com/eclipse-theia/theia/pull/12665)).
-To help we have an [automatic translation](https://github.com/eclipse-theia/theia/actions/workflows/translation.yml) workflow which can be triggered.
-Note that due to the required CI check (`lint`) we will need for force-push the branch that the bot creates to properly trigger CI.
+- Refer to [this example](https://github.com/eclipse-theia/theia/discussions/13314) for guidance.
```

Comment:
We should probably use a more recent example with the updated wording: https://github.com/eclipse-theia/theia/discussions/14547

Reply:
Good point. I updated the link